### PR TITLE
Develop user

### PR DESCRIPTION
**Pull request by formalinfigs**

### CODE REVIEW

File: src/shared/components/photo-item/photo-item.jsx [hunk 1]
```diff
@@ -21,6 +22,11 @@ class PhotoItem extends PureComponent {
             ),
         }).isRequired,
```
Comment (artem-kushal) [inline src/shared/components/photo-item/photo-item.jsx]: photoItem prop dont use on level nesting for one property

File: src/hotel/components/hotel-page/hotel-page.jsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    handleModalClose = () => {
+        this.props.hideModal();
+    };
+
+    render() {
+        if (this.props.hotelInfo && this.props.hotelInfo.id === this.props.match.params.id) {
+            return (
+                <div className="hotel-page">
+                    <HotelItem hotelInfo={this.props.hotelInfo} />
+                    {this.props.user && this.props.user.role !== 'Admin' && (
```
Comment (artem-kushal) [inline src/hotel/components/hotel-page/hotel-page.jsx]: move to role constants

File: src/hotel/components/hotel-page/hotel-page.jsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+                        <div className="hotel-page__booking">
+                            <div className="hotel-page__booking-price">
+                                <span className="hotel-page__booking-price-range">
+                                    ${_.minBy(this.props.hotelInfo.roomTypes, 'cost').cost}-
```
Comment (artem-kushal) [inline src/hotel/components/hotel-page/hotel-page.jsx]: move to container

File: src/hotel/components/modal-booking/modal-booking.jsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+            PropTypes.shape({
+                capacity: PropTypes.number,
+                cost: PropTypes.number,
+                type: PropTypes.string,
+            }),
+        ).isRequired,
+        changeGuestValue: PropTypes.func.isRequired,
+    };
+
+    getSnapshotBeforeUpdate = (prevProps) => {
```
Comment (artem-kushal) [inline src/hotel/components/modal-booking/modal-booking.jsx]: need explanation

File: src/booking/components/booking-page/booking-page.jsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+    render() {
+        const {
+            match: {
+                params: { id },
+            },
+            booking,
+        } = this.props;
+
+        if (booking && booking.id === id) {
```
Comment (artem-kushal) [inline src/booking/components/booking-page/booking-page.jsx]: add isLoading

File: src/admin/actions/actions.js [hunk 6]
```diff
@@ -20,18 +18,46 @@ const showSuccessToast = (message) => {
...
     };
 }
 
-export function addServices(services) {
+// hotelId
+// id
+export function addServices(services, supposedFacilities) {
     return (dispatch) => {
+        const paidFacilities = services.paidFacilities
+            .filter(facility => facility.selectedOption)
```
Comment (artem-kushal) [inline src/admin/actions/actions.js]: move to container

File: src/admin/containers/service-tab-container.js [hunk 7]
```diff
@@ -1,24 +1,69 @@
...
 import ServiceTab from '../components/service-tab/service-tab';
 
-import * as adminActions from '../actions/actions';
+import { SERVICE_ANSWERS, SERVICE_ANSWERS_INDEXES } from '../constants/index';
+
+import * as actions from '../actions/actions';
+
+const getServiceFormInitialValues = (services, supposedFacilities) => {
+    if (supposedFacilities) {
+        const initialValues = {
```
Comment (artem-kushal) [inline src/admin/containers/service-tab-container.js]: need explanation

File: src/booking/components/booking-details/booking-details.jsx [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+    render() {
+        const { handleSubmit, booking, freeFacilities } = this.props;
+
+        const nightsInHotel = moment(booking.dateTo).diff(moment(booking.dateFrom), 'days');
+        const dateLabel = nightsInHotel > 1 ? `${nightsInHotel} nights` : '1 night';
+
+        const timeUnit = moment(booking.dateTo).diff(moment(booking.dateFrom), 'd');
+        const roomPrice = booking.room.cost;
+
+        const detailsPrice = this.props.formValues
```
Comment (artem-kushal) [inline src/booking/components/booking-details/booking-details.jsx]: too many logic in component

File: src/booking/containers/booking-details-container.js [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+import _ from 'lodash';
+
+import BookingDetails from '../components/booking-details/booking-details';
+
+import * as actions from '../actions/actions';
+import * as constants from '../constants';
+
+const getDetailsFormInitialValues = (booking) => {
+    if (booking) {
+        const paidFacilities = booking.room.hotel.services
```
Comment (artem-kushal) [inline src/booking/containers/booking-details-container.js]: need explanation

File: src/main/components/banner/banner.scss [hunk 10]
```diff
@@ -0,0 +1,12 @@
+.banner {
+    background-image: url('/public/images/banner-background.jpg');
+    background-size: cover;
```
[the remark this answers is not in the document]
Comment (formalinfigs) [reply src/main/components/banner/banner.scss]: +

File: src/hotel/components/hotel-page/hotel-page.jsx (hunk 2 above)
Comment (formalinfigs) [reply src/hotel/components/hotel-page/hotel-page.jsx]: +

Comment (formalinfigs) [reply src/hotel/components/hotel-page/hotel-page.jsx]: +

File: src/admin/actions/actions.js (hunk 6 above)
Comment (formalinfigs) [reply src/admin/actions/actions.js]: selector

File: src/booking/components/booking-details/booking-details.jsx (hunk 8 above)
Comment (formalinfigs) [reply src/booking/components/booking-details/booking-details.jsx]: selector

**Pull request merged**